src/platform/datapath_raw_xdp_win.c

```diff
@@ -778,10 +679,20 @@ CxPlatDpRawInterfaceInitialize(
     }
 
     //
-    // Add each queue to a partition (round robin).
+    // Add each queue to the correct partition.
     //
-    for (uint8_t i = 0; i < Interface->QueueCount; i++) {
-        XdpWorkerAddQueue(&Xdp->Partitions[i % Xdp->PartitionCount], &Interface->Queues[i]);
+    for (uint16_t i = 0; i < Interface->QueueCount; i++) {
+        BOOLEAN Found = FALSE;
+        for (uint16_t j = 0; j < Xdp->PartitionCount; j++) {
+            if (Xdp->Partitions[j].Processor == Interface->Queues[i].RssProcessor) {
+                XdpWorkerAddQueue(&Xdp->Partitions[j], &Interface->Queues[i]);
+                Found = TRUE;
+                break;
+            }
+        }
+        if (!Found) {
+            CXPLAT_FRE_ASSERT(FALSE); // TODO - What do we do if there is no partition for this processor?
```

Comment:
@mtfriesen thoughts here? We allow an app to specify the CPUs they want us to run on. What do we do if there is an RSS queue that doesn't have a CPU being run on?

Reply:
I think the question you are asking here is the question I answered in the incorrect location above - a queue without an assigned processor is effectively not a queue.